### DISCUSSION TITLE
react-grid-layout -> pdf

### DISCUSSION THREAD

**Discussion by georgeatavit** [react-grid-layout/react-grid-layout#1544] [Q&A]
Hi. I have this layout (document). I'm using react-grid-layout to render items in different positions and each item has different component in it.
So after completing document there should be an option to generate pdf file. And If you look closer you can see there (page break) component which is simple div with some className where I put css style (page-break-before: always).

I have tried different ways to generate pdf, but it never renders page break correctly.  It seems there’s something wrong with react-grid-layout, because when put same div (div className="page-break") outside layout it works.

Is there any ideas how I can fix this? Or is there any better way to generate pdf from react-grid-layout?

Sorry for english :)

![screencapture-rfloekvyky-localhost-3000-app-documents-f028a7fd-c5c2-4b83-89d6-2404856002c2-2021-08-30-21_13_56](https://user-images.githubusercontent.com/79845532/131380001-9c667889-e291-40de-a1db-fdab31997c8a.png)

**Comment by ghost** [comment]
An answer here would be nice

**Reply by STRML** [reply]
Sorry, there's just not enough information here to go on, and this should be fixable by setting the right CSS rules for print. You'll likely need to remove `position: absolute` on the grid items, but be aware they'll then render in the absolute order they are in the grid - they probably won't be rearranged the way you want them. Hope that helps.
